linkerd/retry/src/replay.rs

```diff
@@ -36,6 +36,12 @@ pub struct ReplayBody<B> {
     replay_trailers: bool,
 }
 
+#[derive(Debug)]
+pub enum MaybeReplay<B, R = ReplayBody<B>> {
+    NoReplay(B),
+    Replay(R),
+}
```

Comment:
Seems like this is really just a `EitherBody`?

Reply:
...i forgot we had that. we can probably get rid of this!

Reply:
> ...i forgot we had that. we can probably get rid of this!

Oh, I misinterpreted your comment; you were saying that this could just be a more generic `EitherBody` type, not that we already had one that we could use here. Yes, it could be --- the only reason it's not generic is that there's a default type param for the replay case, rather than hardcoding `ReplayBody<B>` in the replay case to allow using the same type for the data.

But...I'm going to see if this could just be a `BoxBody`---probably can.